source/configuration/modules/omfile.rst

```diff
@@ -1,434 +1,441 @@
+
+If greater than 0, turns on gzip compression of the output file. The
+higher the number, the better the compression, but also the more CPU
+is required for zipping.
+
+================  ========
+veryRobustZip:    "on|off"
+Default:          "on"
+Available since:  7.3.0
+================  ========
+
+If *zipLevel* is greater than 0,
+then this setting controls if extra headers are written to make the
+resulting file extra hardened against malfunction. 
+If set to *off*, data appended to previously unclean closed files may not be
+accessible without extra tools.
+
+**TODO Note that this risk is usually expected to be bearable, and thus "off" is the default mode.**
+
+The extra headers considerably degrade compression, files with this option set to "on" may be four to five times as large as files processed in "off" mode.
+
+==============  ====================
+flushInterval:  "seconds"
+Default:        "1"
+Legacy:         $OMFileFlushInterval
+==============  ====================
+
+Defines, in seconds, the interval after which unwritten data is flushed.
+
+=============  ===================
+asyncWriting:  "on|off"
+Default:       "off"
+Legacy:        $OMFileASyncWriting
+=============  ===================
+
+if turned *on*, the files will be written in asynchronous mode via a
+separate thread. In that case, double buffers will be used so that
+one buffer can be filled while the other buffer is being written.
+   
+Note that in order to enable *flushInterval*, this parameter must be *on*. Otherwise, the flush interval will be ignored.
+Also note that when flushOnTXEnd is *on* but asyncWriting is *off*, output will only be written when the buffer is full.
+This may take several hours, or even require a rsyslog shutdown. However, a buffer flush can be forced in that case by sending rsyslogd a HUP signal.
+
+=============  ===================
+flushOnTXEnd:  "on|off"
+Default:       "on"
+Legacy:        $OMFileFlushOnTXEnd
+=============  ===================
+
+Omfile has the capability to write output using a buffered writer.
+Disk writes are only done when the buffer is full. So if an error
+happens during that write, data is potentially lost.
+
+In cases where this is unacceptable, set this parameter to *on*.
+Then, data is written at the end of each transaction (for pre-v5 this means after each log
+message) and the usual error recovery thus can handle write errors
+without data loss. 
+Note that this option severely reduces the effect
+of zip compression and should be switched to *off* for that use case.
+
+=============  ===================
+ioBufferSize:  "size"
+Default:       "4K" (KB)
+Legacy:        $OMFileIOBufferSize
+=============  ===================
+
+Size (in KB) of the buffer used to write output data. The larger the
+buffer, the potentially better performance is. The default of 4K is
+quite conservative, it is useful to go up to 64k, and 128K if you
+used gzip compression (then, even higher sizes may make sense).
+   
```

Comment:
It may be worth grouping all the buffersize/flush/timeout/async settings into one section where you have a large block of text that explains how all of them interact, and then has a list of the parameters and their defaults. Right now the explanations are spread across many different parameters but interact with each other